### PR TITLE
Update class-libraries.md

### LINKED ISSUE #12486: [Preview 6] Static content support for razor class libraries

### LINKED ISSUE #12719: Add static content instructions
Add static content instructions

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 37b02665-c8ab-bd91-8057-3b07b4e238d3
* Version Independent ID: e4f176f5-0b94-d532-53d5-cffce437961a
* Content: [Razor components class libraries](https://docs.microsoft.com/en-us/aspnet/core/blazor/class-libraries?view=aspnetcore-3.0&tabs=visual-studio)
* Content Source: [aspnetcore/blazor/class-libraries.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/class-libraries.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #12719
Fixes #12486

### CODE REVIEW

File: aspnetcore/blazor/class-libraries.md [hunk 1]
```diff
@@ -126,3 +126,28 @@ dotnet nuget publish
 ```
 
 When using the `blazorlib` template, static resources are included in the NuGet package. Library consumers automatically receive scripts and stylesheets, so consumers aren't required to manually install the resources. Note that [static assets aren't supported for server-side apps](#static-assets-not-supported-for-server-side-apps), including when a Blazor library (`blazorlib`) is referenced by a server-side app.
+
+## Create a razor class library with static assets
+
+Razor Class Libraries (RCL) frequently require companion static assets that can be to be referenced by the consuming app of the RCL. ASP.NET Core allows creating RCLs that include static assets that are available from a consuming app.
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
Razor class libraries (RCL) frequently require companion static assets that can be referenced by the consuming app of the RCL. ASP.NET Core allows creating RCLs that include static assets that are available to a consuming app.
```

Thus far, we've been going without the proper noun for RCL. If you want to change that, sounds good to me, but it should be updated everywhere in the topic (and perhaps even elsewhere across the docs). If you want, I can take that up later on another PR (or perhaps I'll just update it as I go on the Preview 6 updates coming up shortly).

File: aspnetcore/blazor/class-libraries.md [hunk 2]
```diff
@@ -126,3 +126,28 @@ dotnet nuget publish
...
+
+To include companion assets as part of a razor class library, create a *wwwroot* folder in the class library and include any required files in that folder.
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
To include companion assets as part of a Razor class library, create a *wwwroot* folder in the class library and include any required files in that folder.
```

File: aspnetcore/blazor/class-libraries.md [hunk 3]
```diff
@@ -126,3 +126,28 @@ dotnet nuget publish
...
+
+When packing a razor class library, all companion assets in the *wwwroot* folder are included in the package automatically and are made available to apps referencing the package.
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
When packing a Razor class library, all companion assets in the *wwwroot* folder are included in the package automatically and are made available to apps referencing the package.
```

File: aspnetcore/blazor/class-libraries.md [hunk 4]
```diff
@@ -126,3 +126,28 @@ dotnet nuget publish
...
+
+### Consume content from a referenced razor class library
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
### Consume content from a referenced Razor class library
```

File: aspnetcore/blazor/class-libraries.md [hunk 5]
```diff
@@ -126,3 +126,28 @@ dotnet nuget publish
...
+
+The files included under the *wwwroot* folder of the razor class library are exposed on the consuming app under the prefix `_content/<<libraryname>>/`. The consuming app can reference these assets within script, style, img, etc. tags.
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
The files included in the *wwwroot* folder of the Razor class library are exposed to the consuming app under the prefix `_content/{LIBRARY NAME}/`. The consuming app references these assets by `<script>`, `<style>`, `<img>`, and other HTML tags.
```

File: aspnetcore/blazor/class-libraries.md [hunk 6]
```diff
@@ -126,3 +126,28 @@ dotnet nuget publish
...
+
+### Multi-project development flow
+
+When the app runs:
+
+* The assets stay in their original folders.
+* Any change within the class library *wwwroot* folder is reflected on the app without rebuilding.
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
* Any change within the class library *wwwroot* folder is reflected in the app without rebuilding.
```

File: aspnetcore/blazor/class-libraries.md [hunk 7]
```diff
@@ -126,3 +126,28 @@ dotnet nuget publish
...
+
+At build time, a manifest is produced with all the static web asset locations. The manifest is read at runtime and allows the app to consume the assets from referenced projects and packages.
+
+### Publish
+
+When the app is published, the companion assets from all referenced projects and packages get copied into the *wwwroot* folder of the published app under `_content/<<libraryname>>/`.
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
When the app is published, the companion assets from all referenced projects and packages are copied into the *wwwroot* folder of the published app under `_content/{LIBRARY NAME}/`.
```

**Comment by Rick-Anderson** [comment]
> For the `<<libraryname>>` placeholder, I suggest the syntax that we settled on earlier. 
What's that?

**Comment by guardrex** [comment]
It's no underscores to let localization work (when it isn't code fenced) and braces over angle brackets. For example ...

> The path is */{A PLACEHOLDER SEGMENT}/{ANOTHER SEGMENT}*.

.. or code-fenced ...

> The URL is `http://localhost:5000/{URL SEGMENT}`.

Not localized in the URL case, but consistent.

I can get the convo if you want to see it ... it's out there in GH issue/PR history somewhere.

**Comment by guardrex** [comment]
:smile: I was close ...

https://github.com/aspnet/AspNetCore.Docs/pull/8014

The "no underscores" thing came later ... on other PRs. It's obvious that without them the machine localization has a good shot at getting them translated. Turns out that on that PR it didn't matter: The stuff is code-fenced on that one.

File: aspnetcore/blazor/class-libraries.md [hunk 8]
```diff
@@ -126,3 +126,28 @@ dotnet nuget publish
 ```
 
 When using the `blazorlib` template, static resources are included in the NuGet package. Library consumers automatically receive scripts and stylesheets, so consumers aren't required to manually install the resources. Note that [static assets aren't supported for server-side apps](#static-assets-not-supported-for-server-side-apps), including when a Blazor library (`blazorlib`) is referenced by a server-side app.
+
+## Create a razor class library with static assets
```
Comment (pranavkm) [inline aspnetcore/blazor/class-libraries.md]: cc @javiercn

**Comment by Rick-Anderson** [comment]
So what do you want to use?

File: aspnetcore/blazor/class-libraries.md (hunk 8 above)
Comment (javiercn) [reply aspnetcore/blazor/class-libraries.md]: It's an asp.net core feature, not a blazor specific feature

**Comment by javiercn** [comment]
> So what do you want to use For the `<<libraryname>>` placeholder ?

what do you mean?

For example if your library is called MySuperAWESOMELibrary the content ends up in `_content/mysuperawesomelibrary`

**Comment by guardrex** [comment]
I've been going with braces and no underscores recently. Truly tho ... looking across the whole doc set, there is no consistency AFAICT. It worked well with braces in that PR because the reader's comment was a good one: angles there get confused with C#. In most cases tho, that's not really a problem. Let me know you go a different way, and I'll massage things that direction.

**Comment by guardrex** [comment]
> For example if your library is called MySuperAWESOMELibrary the content ends up in _content/mysuperawesomelibrary

Correct ... it's a placeholder. IMO, we should have a consistent way of showing user-supplied content for placeholders in various strings and values.

**Comment by javiercn** [comment]
> Correct ... it's a placeholder. IMO, we should have a consistent way of showing user-supplied content for placeholders in various strings and values.

I'll leave those decisions to your best judgement

**Comment by guardrex** [comment]
Yes ... indeed ... I yield to the team to let me know how they'd like to do it.

**Comment by javiercn** [comment]
@danroth27 Can you help here?

**Comment by danroth27** [comment]
@guardrex's proposal sounds fine to me.

**Pull request merged**